hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -3632,7 +3633,7 @@ TermIndex installCheckpoint(String leaderId, Path checkpointLocation,
       keyManager.stop();
       stopSecretManager();
       stopTrashEmptier();
-
+      omSnapshotManager.getSnapshotCache().invalidateAll();
```

Comment:
I see it solves the problem but ideally we should only invalidate cache when [OmMetadataManager was stopped](https://github.com/apache/ozone/blob/a66e43e2838f087229cb4b0fac1e1dc969ab80dd/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java#L3743) because when `OmMetadataManager` gets stopped we load [everything including cache fresh](https://github.com/apache/ozone/blob/a66e43e2838f087229cb4b0fac1e1dc969ab80dd/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone)/om/OzoneManager.java#L3736C9-L3736C22).

Reply:
Had a offline discussion and problem is that in case of reloading OmMetadataManager, previous [snapshotCache](https://github.com/apache/ozone/blob/5a6f08ce48eb7053894596a7bb2d3993072f8c8a/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java#L116) object may/may not get GC'ed and new snapshotCache object get created. If previous snapshotCache is not GC'ed and has opened RocksDB instance, it will keep deleting file because of RocksDB async delete (delete command was executed by previous SSTFilteringService run). New cache will try to open RocksDB instance based on new manifest and will fail because file is missing. 

Based on logs: On followed node,
1. File 000077.sst was deleted by SSTFilteringService at hemant_sst_filtering.log line 20972.
1. File 000077.sst was copied from candidate dir to snapshot db dir at hemant_sst_filtering.log line 30871
1. Background services including SSTFilteringService get started at hemant_sst_filtering.log line 31062
1. Later SSTFilteringService run fails because file doesn't exist. at hemant_sst_filtering.log line 31111-31112.
Which was the overall `testInstallIncrementalSnapshot`.

Attached the log file for details: [testInstallIncrementalSnapshot.log](https://github.com/apache/ozone/files/12033012/hemant_sst_filtering.log)

Reply:
@hemantk-12  So this all sounds good to me, but one thing that isn't clear.  It sounds, from the explanation above, that we believe, somewhere between steps 2 and 4, the old cache entry deleted 77.sst a second time.

Is that a correct understanding of your theory?

Is there evidence of that second delete in the logs?

Reply:
> Is that a correct understanding of your theory?

Yes, that's the correct. I don't have logs for that but Swami might have because he enabled native RocksDB logs. @swamirishi, please add the logs either here or in description if you still have.

Reply:
I have attached the log in the jira.